pkg/clusteragent/admission/mutate/agent_sidecar/util_test.go

```diff
@@ -108,3 +109,228 @@ func TestApplyProviderOverrides(t *testing.T) {
 		})
 	}
 }
+
+func TestApplyProfileOverrides(t *testing.T) {
+	mockConfig := config.Mock(t)
+
+	tests := []struct {
+		name              string
+		profilesJSON      string
+		baseContainer     *corev1.Container
+		expectedContainer *corev1.Container
+	}{
+		{
+			name:              "nil container should be skipped",
```

Comment:
I am handling the case of `nil` pod in the [mutation function](https://github.com/DataDog/datadog-agent/blob/15cf63aa1e85d62bb6cb5b18824b98da77facb25/pkg/clusteragent/admission/mutate/agent_sidecar/agent_sidecar.go#L27-L29).

I think this should be enough.

Reply:
if `nil` should never happen, I'd be in favor of logging/throwing error (kind of assertion) because caller code may change and pass a `nil`. In that case it's better to fail fast so it can be debugged at this line. If we skip and overrides don't get applied, one has to look into config, unmarshaling, and several if else statements to narrow down the root cause.

Reply:
agreed 👍